cmd/auth/auth.go

```diff
@@ -68,11 +69,20 @@ var AuthCmd = &cobra.Command{
 			password = strings.TrimSpace(string(bytePassword))
 		}
 
+		baseURL := ""
+		if backend == "llama" {
```

Comment:
I'm not very comfortable to put backend-specific code in auth, however, in order to avoid this I think some refactor would be needed (for instance,  ask for backend-specific settings programmatically) to the interfaces. I didn't felt should be part of this PR, so I went with a simple approach. 

Let me know your feedback, I think we can generalize that easier later and keep this PR's scope just to the local backend

Reply:
I totally agree.

I think we should refactor this part of the code to make it more flexible to adding new backends, especially since @arbreezy is working on adding a new backend it seems.

Reply:
Are you okay holding off a little until that PR is in @mudler ? I don't mind either way.

Reply:
I'm fine either ways, at the end is your call!  I'm just a little bit concerned this one can start to grow too much out of scope and lead to conflicts. Is there a branch somewhere already? I could rebase on top of it and start to play with it

Reply:
afterthought maybe merging this in before might be actually helpful when refactoring providers - it gives a concrete example on what other providers might need to support. But I'm fine as well rebasing my work on top of the refactoring after that's being merged. Just cc'ing @arbreezy so he is aware of what's needed for llama here :arrow_up:

Reply:
hello @mudler @AlexsJones  !
I 've added a draft [PR](https://github.com/k8sgpt-ai/k8sgpt/pull/297) with a first iteration of introducing a configuration interface and also adding llama and azure open ai.
I just wanted to get all changes in a single place for this work, eventually this draft PR will break in individual components (config interface, azure open ai, llama, etc)

Reply:
hey @arbreezy ! thanks for taking a stab at it ! It looks good to me!

maybe I'd just split the llama changes from it, so can be introduced separately here, what do you think? otherwise it doesn't have documentation with the PR - or I can just add documentation here. I have no issues with it, just let me know!

Reply:
Yes, it's only fair to leave llama to you , just wanted to demonstrate how will look like, if everyone is happy I will close the draft PR and introduce two smaller PRs around config changes and azure open ai.

Reply:
have also some updates on my side due to support of GPT4ALL-J, will update based off from your work!

Reply:
awesome, I will find time tomorrow to create the first PR which introduces config interface (I won't include any llama related code)